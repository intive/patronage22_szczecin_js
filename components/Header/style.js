import styled from 'styled-components'

export const HeaderContainer = styled.header`
  position: sticky;
  left: 0;
  top: 0;
  width: 100%;
  height: 4.5rem;
  background: #FFFFFF;
  `

export const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  padding: 0 5%;

  @media (min-width: 1440px) {
    padding: 0 2.5rem;
  }
`

export const HeaderTitle = styled.a`
  display: flex;
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  text-decoration: none;
  color: inherit;
  `

export const HeaderTitleText = styled.p`
  margin-left: 0.5rem;
  color: #222222;
`
