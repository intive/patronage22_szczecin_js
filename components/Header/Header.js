import { HeaderContainer, HeaderWrapper, HeaderTitle, HeaderTitleText } from './style'
import Icon from '../Icon/Icon'
import Link from 'next/link'

export default function Header (props) {
  return (
    <HeaderContainer>
      <HeaderWrapper>
        <Link href='/'>
          <HeaderTitle>
            <Icon name='crop_original' />
            <HeaderTitleText>{props.title}</HeaderTitleText>
          </HeaderTitle>
        </Link>
      </HeaderWrapper>
    </HeaderContainer>
  )
}
