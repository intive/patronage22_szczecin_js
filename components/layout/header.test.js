/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import Header from './Header'

describe('Header', () => {
  it('should render Header', () => {
    const header = render(<Header />)
    expect(header).toMatchSnapshot()
  })
})
