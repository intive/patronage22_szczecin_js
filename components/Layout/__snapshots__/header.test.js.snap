// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<div>
  <header
    class="sc-bdvvtL fFkdhI"
  >
    <div
      class="sc-gsDKAQ LTTzv"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 5V19H5V5H19ZM19 3H5C3.9 3 3 3.9 3 5V19C3 20.1 3.9 21 5 21H19C20.1 21 21 20.1 21 19V5C21 3.9 20.1 3 19 3ZM14.14 11.86L11.14 15.73L9 13.14L6 17H18L14.14 11.86Z"
          fill="black"
        />
      </svg>
      <p
        class="sc-dkPtRN dzbwyK"
      >
        My Retro Board
      </p>
    </div>
  </header>
</div>
`;
